crates/egui/src/widgets/text_edit/builder.rs

```diff
@@ -671,8 +671,7 @@ impl<'t> TextEdit<'t> {
                     let primary_cursor_rect =
                         cursor_rect(galley_pos, &galley, &cursor_range.primary, row_height);
 
-                    let is_fully_visible = ui.clip_rect().contains_rect(rect); // TODO: remove this HACK workaround for https://github.com/emilk/egui/issues/1531
-                    if (response.changed || selection_changed) && !is_fully_visible {
+                    if response.changed || selection_changed {
```

Comment:
Nice to get rid of this hack! Can we get rid of it in `crates/egui/src/text_selection/label_text_selection.rs` too?